Unit test failed at commit 8fca348 on v1.5

## Describe the bug

Hi there! I'm using Qt 5.15.x mingw810_64 but I failed some tests.

## Environment

- My PC
    - Windows 10 2004 Pro 10.0.19041.685
    - CMD
    - Qt
        - Qt 5.15.0 (MinGW 8.1.0 64-bit)
        - IFW 4.0.1
- GitHub Actions ([virtual-environments_Windows2019-Readme.md at win19_20201210.0 · actions_virtual-environments](https://github.com/actions/virtual-environments/blob/win19/20201210.0/images/win/Windows2019-Readme.md))
    - Windows Server 2019 Datacenter 10.0.17763.1637
    - PowerShell 7.1.0
    - Qt
        - Qt 5.15.0 / 5.15.1 / 5.15.2 (MinGW 8.1.0 64-bit)
        - IFW 4.0.1

## To Reproduce

Steps to reproduce the behavior:
1. Install Qt 5.15.0 (MinGW 8.1.0 64-bit) and IFW 4.0.1 (use Qt Maintenance Tool on my PC, or [install-qt-action](https://github.com/jurplel/install-qt-action) on GitHub Actions).
2. Install CQtDeployer 1.4.7 with `WindowsOfflineInstaller.exe`.
3. Clone this project and checkout to the commit 8fca34848b2c79caa4582502335518f39dfb49e4 .
4. Execute `qmake.exe -r`, `mingw32-make.exe` and `mingw32-make.exe test` in `CQtDeployer` source code folder.
5. See error in stdout and `CQtDeployer\buildLog.log`.

## Expected behavior

Pass all tests.

## Error info

### On my PC

#### Qt 5.15.0

stdout:

```console
mingw32-make: *** [Makefile:868: test] Error 1
```

Full content of `CQtDeployer\buildLog.log`:

<details>
<summary>buildLog.log</summary>

```
********* Start testing of deploytest *********
Config: Using QtTest library 5.15.0, Qt 5.15.0 (x86_64-little_endian-llp64 shared (dynamic) release build; by GCC 8.1.0)
PASS   : deploytest::initTestCase()
PASS   : deploytest::testDeployTarget()
PASS   : deploytest::testStrip()
PASS   : deploytest::testExtractLib()
PASS   : deploytest::testRelativeLink()
PASS   : deploytest::testCheckQt()
PASS   : deploytest::testQmlExtrct()
PASS   : deploytest::testSetTargetDir()
PASS   : deploytest::testZip()
PASS   : deploytest::costomScript()
PASS   : deploytest::testDistroStruct()
PASS   : deploytest::testOverwrite()
PASS   : deploytest::testextraData()
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/qml/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/QtWidgetsProject.exe not exits in qt Dir"
PASS   : deploytest::testQt()
PASS   : deploytest::testWebEngine()
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/qml/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/Dstro2/translations/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/Dstro2/qml/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/Dstro2/translations/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testConfFile()
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/package2/ZzZ/q/and/q/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/package2/ZzZ/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/lolTr/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/package2/ZzZ/translations/qtdeclarative_tr.qm not exits in qt Dir"
PASS   : deploytest::testPackages()
PASS   : deploytest::testClear()
QWARN  : deploytest::testIgnore() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testIgnore()
PASS   : deploytest::testLibDir()
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testExtraPlugins()
PASS   : deploytest::testTargetDir()
QSYSTEM: deploytest::testSystemLib() "Missing file : D:/CQtDeployer/DistributionKit/profapi.dll"
QWARN  : deploytest::testSystemLib() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testSystemLib() "File : D:/CQtDeployer/DistributionKit/API-MS-Win-devices-config-L1-1-0.dll not exits in qt Dir"
QWARN  : deploytest::testSystemLib() "File : D:/CQtDeployer/DistributionKit/API-MS-Win-devices-config-L1-1-1.dll not exits in qt Dir"
QSYSTEM: deploytest::testSystemLib() "Missing file : D:/CQtDeployer/DistributionKit/bcryptprimitives.dll"
FAIL!  : deploytest::testSystemLib() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/q/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/q/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testOutDirs()
PASS   : deploytest::testMSVC()
PASS   : deploytest::testEmptyParamsString()
PASS   : deploytest::testQIF()
PASS   : deploytest::testQIFMulti()
PASS   : deploytest::testQIFCustom()
PASS   : deploytest::testZIP()
PASS   : deploytest::testZIPMulti()
PASS   : deploytest::testDEB()
PASS   : deploytest::testDEBMulti()
PASS   : deploytest::testDEBCustom()
PASS   : deploytest::testMultiPacking()
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
PASS   : deploytest::testInit()
PASS   : deploytest::testDependencyMap()
PASS   : deploytest::testQmlScaner()
PASS   : deploytest::testPrefix()
PASS   : deploytest::testallowEmptyPackages()
PASS   : deploytest::testEmptyPackages()
QWARN  : deploytest::customTest() QSettings::setValue: Empty key passed
PASS   : deploytest::customTest()
PASS   : deploytest::cleanupTestCase()
Totals: 42 passed, 1 failed, 0 skipped, 0 blacklisted, 303820ms
********* Finished testing of deploytest *********
```

</details>

## Error info

### On GitHub Actions

#### Qt 5.15.0

stdout:

```console
mingw32-make: *** [Makefile:868: test] Error 5
```

Full content of `CQtDeployer\buildLog.log`:

<details>
<summary>buildLog.log</summary>

```
********* Start testing of deploytest *********
Config: Using QtTest library 5.15.0, Qt 5.15.0 (x86_64-little_endian-llp64 shared (dynamic) release build; by GCC 8.1.0)
PASS   : deploytest::initTestCase()
PASS   : deploytest::testDeployTarget()
PASS   : deploytest::testStrip()
PASS   : deploytest::testExtractLib()
PASS   : deploytest::testRelativeLink()
PASS   : deploytest::testCheckQt()
PASS   : deploytest::testQmlExtrct()
PASS   : deploytest::testSetTargetDir()
PASS   : deploytest::testZip()
PASS   : deploytest::costomScript()
PASS   : deploytest::testDistroStruct()
PASS   : deploytest::testOverwrite()
PASS   : deploytest::testextraData()
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/qml/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/QtWidgetsProject.exe not exits in qt Dir"
PASS   : deploytest::testQt()
PASS   : deploytest::testWebEngine()
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/qml/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/Dstro2/qml/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/Dstro2/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testConfFile() "File : D:/CQtDeployer/DistributionKit/Dstro2/translations/qtdeclarative_tr.qm not exits in qt Dir"
PASS   : deploytest::testConfFile()
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/package2/ZzZ/translations/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/lolTr/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/package2/ZzZ/q/and/q/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testPackages() "File : D:/CQtDeployer/DistributionKit/package2/ZzZ/translations/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testPackages()
PASS   : deploytest::testClear()
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libgcc_s_seh-1.dll"
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libwinpthread-1.dll"
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libstdc++-6.dll"
FAIL!  : deploytest::testIgnore() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QWARN  : deploytest::testIgnore() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testLibDir()
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QSYSTEM: deploytest::testExtraPlugins() "added unnecessary file : D:/CQtDeployer/DistributionKit/libpq.dll"
FAIL!  : deploytest::testExtraPlugins() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QSYSTEM: deploytest::testExtraPlugins() "added unnecessary file : D:/CQtDeployer/DistributionKit/libpq.dll"
FAIL!  : deploytest::testExtraPlugins() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testTargetDir()
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-namedpipe-l1-2-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-kernel32-legacy-l1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-com-private-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-processsecurity-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-localization-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-ansi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-string-obsolete-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-obsolete-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-memory-l1-1-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-robuffer-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-shutdown-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-stringloader-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l1-2-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/API-MS-Win-Core-Kernel32-Private-L1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-string-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-normalization-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-registration-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-roparameterizediid-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-2.dll"
FAIL!  : deploytest::testSystemLib() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-namedpipe-l1-2-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-com-private-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-kernel32-legacy-l1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-processsecurity-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-localization-l2-1-0.dll"
QWARN  : deploytest::testSystemLib() "File : D:/CQtDeployer/DistributionKit/translations/qtbase_tr.qm not exits in qt Dir"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-ansi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-string-obsolete-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-obsolete-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-memory-l1-1-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-robuffer-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-shutdown-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-stringloader-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l1-2-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/API-MS-Win-Core-Kernel32-Private-L1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-registration-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-string-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-normalization-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-security-cpwl-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-roparameterizediid-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-2.dll"
FAIL!  : deploytest::testSystemLib() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/q/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtbase_tr.qm not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtdeclarative_tr.qm not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/q/QtQuick/Controls.2/designer/InsetSection.qml not exits in qt Dir"
QWARN  : deploytest::testOutDirs() "File : D:/CQtDeployer/DistributionKit/lolTr/qtbase_tr.qm not exits in qt Dir"
PASS   : deploytest::testOutDirs()
PASS   : deploytest::testMSVC()
PASS   : deploytest::testEmptyParamsString()
PASS   : deploytest::testQIF()
PASS   : deploytest::testQIFMulti()
PASS   : deploytest::testQIFCustom()
PASS   : deploytest::testZIP()
PASS   : deploytest::testZIPMulti()
PASS   : deploytest::testDEB()
PASS   : deploytest::testDEBMulti()
PASS   : deploytest::testDEBCustom()
PASS   : deploytest::testMultiPacking()
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
PASS   : deploytest::testInit()
PASS   : deploytest::testDependencyMap()
PASS   : deploytest::testQmlScaner()
PASS   : deploytest::testPrefix()
PASS   : deploytest::testallowEmptyPackages()
PASS   : deploytest::testEmptyPackages()
QWARN  : deploytest::customTest() QSettings::setValue: Empty key passed
PASS   : deploytest::customTest()
PASS   : deploytest::cleanupTestCase()
Totals: 40 passed, 5 failed, 0 skipped, 0 blacklisted, 214243ms
********* Finished testing of deploytest *********
```

## Error info

### On GitHub Actions

#### Qt 5.15.1

stdout:

```console
mingw32-make: *** [Makefile:868: test] Error 5
```

Full content of `CQtDeployer\buildLog.log`:

<details>
<summary>buildLog.log</summary>

```
********* Start testing of deploytest *********
Config: Using QtTest library 5.15.1, Qt 5.15.1 (x86_64-little_endian-llp64 shared (dynamic) release build; by GCC 8.1.0)
PASS   : deploytest::initTestCase()
PASS   : deploytest::testDeployTarget()
PASS   : deploytest::testStrip()
PASS   : deploytest::testExtractLib()
PASS   : deploytest::testRelativeLink()
PASS   : deploytest::testCheckQt()
PASS   : deploytest::testQmlExtrct()
PASS   : deploytest::testSetTargetDir()
PASS   : deploytest::testZip()
PASS   : deploytest::costomScript()
PASS   : deploytest::testDistroStruct()
PASS   : deploytest::testOverwrite()
PASS   : deploytest::testextraData()
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/QtWidgetsProject.exe not exits in qt Dir"
PASS   : deploytest::testQt()
PASS   : deploytest::testWebEngine()
PASS   : deploytest::testConfFile()
PASS   : deploytest::testPackages()
PASS   : deploytest::testClear()
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libgcc_s_seh-1.dll"
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libwinpthread-1.dll"
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libstdc++-6.dll"
FAIL!  : deploytest::testIgnore() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
PASS   : deploytest::testLibDir()
QSYSTEM: deploytest::testExtraPlugins() "added unnecessary file : D:/CQtDeployer/DistributionKit/libpq.dll"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
FAIL!  : deploytest::testExtraPlugins() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QSYSTEM: deploytest::testExtraPlugins() "added unnecessary file : D:/CQtDeployer/DistributionKit/libpq.dll"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
FAIL!  : deploytest::testExtraPlugins() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
PASS   : deploytest::testTargetDir()
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-string-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-ansi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-obsolete-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-namedpipe-l1-2-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-memory-l1-1-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-processsecurity-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-roparameterizediid-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-com-private-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-robuffer-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-kernel32-legacy-l1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-localization-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-normalization-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l1-2-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-shutdown-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-registration-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-stringloader-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/API-MS-Win-Core-Kernel32-Private-L1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-string-obsolete-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-1.dll"
FAIL!  : deploytest::testSystemLib() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-string-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-ansi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-obsolete-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-namedpipe-l1-2-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-memory-l1-1-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-security-cpwl-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-processsecurity-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-roparameterizediid-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-com-private-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-robuffer-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-kernel32-legacy-l1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-localization-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-normalization-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l1-2-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-shutdown-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-registration-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-stringloader-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/API-MS-Win-Core-Kernel32-Private-L1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-string-obsolete-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-1.dll"
FAIL!  : deploytest::testSystemLib() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
PASS   : deploytest::testOutDirs()
PASS   : deploytest::testMSVC()
PASS   : deploytest::testEmptyParamsString()
PASS   : deploytest::testQIF()
PASS   : deploytest::testQIFMulti()
PASS   : deploytest::testQIFCustom()
PASS   : deploytest::testZIP()
PASS   : deploytest::testZIPMulti()
PASS   : deploytest::testDEB()
PASS   : deploytest::testDEBMulti()
PASS   : deploytest::testDEBCustom()
PASS   : deploytest::testMultiPacking()
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
PASS   : deploytest::testInit()
PASS   : deploytest::testDependencyMap()
PASS   : deploytest::testQmlScaner()
PASS   : deploytest::testPrefix()
PASS   : deploytest::testallowEmptyPackages()
PASS   : deploytest::testEmptyPackages()
QWARN  : deploytest::customTest() QSettings::setValue: Empty key passed
PASS   : deploytest::customTest()
PASS   : deploytest::cleanupTestCase()
Totals: 40 passed, 5 failed, 0 skipped, 0 blacklisted, 202645ms
********* Finished testing of deploytest *********
```

## Error info

### On GitHub Actions

#### Qt 5.15.2

stdout:

```console
mingw32-make: *** [Makefile:868: test] Error 5
```

Full content of `CQtDeployer\buildLog.log`:

<details>
<summary>buildLog.log</summary>

```
********* Start testing of deploytest *********
Config: Using QtTest library 5.15.2, Qt 5.15.2 (x86_64-little_endian-llp64 shared (dynamic) release build; by GCC 8.1.0), windows server2016
PASS   : deploytest::initTestCase()
PASS   : deploytest::testDeployTarget()
PASS   : deploytest::testStrip()
PASS   : deploytest::testExtractLib()
PASS   : deploytest::testRelativeLink()
PASS   : deploytest::testCheckQt()
PASS   : deploytest::testQmlExtrct()
PASS   : deploytest::testSetTargetDir()
PASS   : deploytest::testZip()
PASS   : deploytest::costomScript()
PASS   : deploytest::testDistroStruct()
PASS   : deploytest::testOverwrite()
PASS   : deploytest::testextraData()
QWARN  : deploytest::testQt() "File : D:/CQtDeployer/DistributionKit/QtWidgetsProject.exe not exits in qt Dir"
PASS   : deploytest::testQt()
PASS   : deploytest::testWebEngine()
PASS   : deploytest::testConfFile()
PASS   : deploytest::testPackages()
PASS   : deploytest::testClear()
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libgcc_s_seh-1.dll"
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libstdc++-6.dll"
QSYSTEM: deploytest::testIgnore() "added unnecessary file : D:/CQtDeployer/DistributionKit/libwinpthread-1.dll"
FAIL!  : deploytest::testIgnore() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
PASS   : deploytest::testLibDir()
QSYSTEM: deploytest::testExtraPlugins() "added unnecessary file : D:/CQtDeployer/DistributionKit/libpq.dll"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
FAIL!  : deploytest::testExtraPlugins() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QSYSTEM: deploytest::testExtraPlugins() "added unnecessary file : D:/CQtDeployer/DistributionKit/libpq.dll"
QWARN  : deploytest::testExtraPlugins() "File : D:/CQtDeployer/DistributionKit/plugins/sqldrivers/qsqlmysql.dll not exits in qt Dir"
FAIL!  : deploytest::testExtraPlugins() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
PASS   : deploytest::testTargetDir()
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-roparameterizediid-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-kernel32-legacy-l1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-ansi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-robuffer-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-obsolete-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l1-2-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-shutdown-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-memory-l1-1-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-normalization-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-processsecurity-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-localization-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-string-obsolete-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-registration-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-string-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-com-private-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-namedpipe-l1-2-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/API-MS-Win-Core-Kernel32-Private-L1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-stringloader-l1-1-0.dll"
FAIL!  : deploytest::testSystemLib() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-roparameterizediid-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-error-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-ansi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-kernel32-legacy-l1-1-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-obsolete-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-robuffer-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l1-2-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-security-cpwl-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-shutdown-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-memory-l1-1-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-normalization-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-localization-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-processsecurity-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-psapi-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-string-obsolete-l1-1-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-com-private-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-string-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-winrt-registration-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-namedpipe-l1-2-1.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-2.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-sysinfo-l1-2-3.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-console-l2-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/api-ms-win-core-stringloader-l1-1-0.dll"
QSYSTEM: deploytest::testSystemLib() "added unnecessary file : D:/CQtDeployer/DistributionKit/API-MS-Win-Core-Kernel32-Private-L1-1-2.dll"
FAIL!  : deploytest::testSystemLib() 'false' returned FALSE. (runTestParams fail)
tst_deploytest.cpp(1474) : failure location
PASS   : deploytest::testOutDirs()
PASS   : deploytest::testMSVC()
PASS   : deploytest::testEmptyParamsString()
PASS   : deploytest::testQIF()
PASS   : deploytest::testQIFMulti()
PASS   : deploytest::testQIFCustom()
PASS   : deploytest::testZIP()
PASS   : deploytest::testZIPMulti()
PASS   : deploytest::testDEB()
PASS   : deploytest::testDEBMulti()
PASS   : deploytest::testDEBCustom()
PASS   : deploytest::testMultiPacking()
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
QWARN  : deploytest::testInit() QSettings::setValue: Empty key passed
PASS   : deploytest::testInit()
PASS   : deploytest::testDependencyMap()
PASS   : deploytest::testQmlScaner()
PASS   : deploytest::testPrefix()
PASS   : deploytest::testallowEmptyPackages()
PASS   : deploytest::testEmptyPackages()
QWARN  : deploytest::customTest() QSettings::setValue: Empty key passed
PASS   : deploytest::customTest()
PASS   : deploytest::cleanupTestCase()
Totals: 40 passed, 5 failed, 0 skipped, 0 blacklisted, 203370ms
********* Finished testing of deploytest *********
```

--- a/Deploy/envirement.cpp
+++ b/Deploy/envirement.cpp
@@ -113,7 +113,7 @@ QString Envirement::concatEnv() const {
     }
 
     QString result = *_dataEnvironment.begin();
-    for (auto i: _dataEnvironment) {
+    for (const auto &i: _dataEnvironment) {
         result += (DeployCore::getEnvSeparator() + i);
     }
 

--- a/Deploy/envirement.cpp
+++ b/Deploy/envirement.cpp
@@ -98,7 +98,7 @@ bool Envirement::inThisEnvirement(const QString &file) const {
         return _dataEnvironment.contains(PathUtils::fixPath(info.absolutePath()));
     }
 
-    return _dataEnvironment.contains(PathUtils::fixPath(file));
+    return _dataEnvironment.contains(PathUtils::fixPath(info.absoluteFilePath()));
 
 }
 

--- a/Deploy/ignorerule.cpp
+++ b/Deploy/ignorerule.cpp
@@ -9,7 +9,7 @@
 #include <quasarapp.h>
 
 bool IgnoreRule::checkOnlytext(const QString &lib) {
-    for (auto ignore : _data) {
+    for (const auto &ignore : qAsConst(_data)) {
         if (lib.contains(ignore.label)) {
             return true;
         }
@@ -27,7 +27,7 @@ void IgnoreRule::addRule(const IgnoreData &rule) {
 }
 
 bool IgnoreRule::check(const LibInfo &info, const QString& ignoreLabel) const {
-    if (info.fullPath().contains(ignoreLabel, Qt::CaseInsensitive)) {
+    if (info.fullPath().contains(ignoreLabel, ONLY_WIN_CASE_INSENSIATIVE)) {
         QuasarAppUtils::Params::log(info.fullPath() + " ignored by filter" + ignoreLabel);
         return true;
     }

--- a/UnitTests/modulesqt513.cpp
+++ b/UnitTests/modulesqt513.cpp
@@ -20,7 +20,7 @@ ModulesQt513::~ModulesQt513() {
 QSet<QString> ModulesQt513::ignoreFilter(const QSet<QString> &input, const QString &filter) const {
     QSet<QString> res;
     for (auto& val : input) {
-        if (!val.contains(filter)) {
+        if (!val.contains(filter, ONLY_WIN_CASE_INSENSIATIVE)) {
             res.insert(val);
         }
     }

--- a/UnitTests/testutils.cpp
+++ b/UnitTests/testutils.cpp
@@ -35,7 +35,7 @@ QSet<QString> TestUtils::getTree(const QString &path, int limit, int depch) {
 
     QDir dir(info.absoluteFilePath());
     auto list = dir.entryInfoList(QDir::AllEntries | QDir::NoDotAndDotDot);
-    for (const auto &i: list) {
+    for (const auto &i: qAsConst(list)) {
         result.unite(getTree(i.absoluteFilePath(), limit, depch + 1));
     }
 

--- a/UnitTests/tst_deploytest.cpp
+++ b/UnitTests/tst_deploytest.cpp
@@ -33,32 +33,28 @@
 // add necessary includes here
 
 
-const QString TestBinDir = TEST_BIN_DIR;
-const QString TestQtDir = QT_BASE_DIR;
+static const QString TestBinDir = TEST_BIN_DIR;
+static const QString TestQtDir = QT_BASE_DIR;
 
 class deploytest : public QObject
 {
     Q_OBJECT
 
 private:
-    QHash<QString, QSet<QString>> filesTree;
+    QSet<QString> filesTree;
 
     bool runProcess(const QString& DistroPath,
                     const QString& filename,
                     const QString &qt = "");
     QStringList getFilesFromDir(const QString& dir);
 
-    QSet<QString> getFilesTree(const QStringList& keys = {});
-
     void runTestParams(QStringList list,
                        QSet<QString> *tree = nullptr,
-                       const QStringList &checkableKeys = {},
                        bool noWarnings = false,
                        bool onlySize = false,
                        exitCodes exitCode = exitCodes::Good);
 
     void checkResults(const QSet<QString> &tree,
-                      const QStringList &checkagbleKeys,
                       bool noWarnings,
                       bool onlySize = false);
 public:
@@ -245,53 +241,19 @@ QStringList deploytest::getFilesFromDir(const QString &path) {
     return res;
 }
 
-QSet<QString> deploytest::getFilesTree(const QStringList &keys) {
-    QSet<QString> result;
-
-    if (keys.isEmpty()) {
-        return filesTree["all"];
-    }
-
-    for (const auto& i: keys) {
-        result += filesTree[i];
-    }
-
-    return result;
-}
-
 deploytest::deploytest() {
 
     qputenv("QTEST_FUNCTION_TIMEOUT", "1800000");
 
     TestUtils utils;
 
-    auto tempTree = utils.getTree(TestQtDir);
-    for (const QString &i: tempTree) {
-        filesTree["Qt"].insert(QFileInfo(i).fileName());
+    const QStringList pathList = QProcessEnvironment::systemEnvironment().
+            value("PATH").split(DeployCore::getEnvSeparator());
+
+
+    for (const auto& path: pathList) {
+        filesTree += utils.getTree(path, 1);
     }
-
-    tempTree = utils.getTree("/lib", 5);
-    for (const QString &i: tempTree) {
-        filesTree["/lib"].insert(QFileInfo(i).fileName());
-    }
-
-    tempTree = utils.getTree("/usr/lib", 5);
-    for (const QString &i: tempTree) {
-        filesTree["/usr/lib"].insert(QFileInfo(i).fileName());
-    }
-
-    tempTree = utils.getTree("C:/windows/system32", 2);
-    for (const QString &i: tempTree) {
-        filesTree["C:/windows/system32"].insert(QFileInfo(i).fileName());
-    }
-
-    QSet<QString> all;
-    for (const auto& i : filesTree) {
-        all.unite(i);
-    }
-    filesTree["all"] = all;
-
-
 }
 
 int deploytest::generateLib(const QString &paath)
@@ -603,7 +565,7 @@ void deploytest::testQIF() {
                    "qif", "qifFromSystem",
                    "-qifStyle", "quasar",
                    "-qifBanner", TestBinDir + "/../../res/CQtDeployer_banner_web.png",
-                   "-qifLogo", TestBinDir + "/../../res/CQtDeployer defaultIcon_web.png"}, &comapareTree, {}, true);
+                   "-qifLogo", TestBinDir + "/../../res/CQtDeployer defaultIcon_web.png"}, &comapareTree, true);
 
 
 
@@ -654,7 +616,7 @@ void deploytest::testQIFMulti() {
                    "-qmlOut", "/q",
                    "-qmlDir", "package2;" + TestBinDir + "/../TestQMLWidgets",
                    "-targetPackage", packageString,
-                   "qif", "qifFromSystem"}, &comapareTreeMulti, {}, true);
+                   "qif", "qifFromSystem"}, &comapareTreeMulti, true);
 }
 
 void deploytest::testQIFCustom() {
@@ -683,7 +645,7 @@ void deploytest::testQIFCustom() {
                    "-qmlDir", TestBinDir + "/../TestQMLWidgets",
                    "-qif", TestBinDir + "/../../UnitTests/testRes/QIFCustomTemplate",
                    "-name", "org.qtproject.ifw.example.stylesheet",
-                   "qifFromSystem"}, &comapareTreeCustom, {}, true);
+                   "qifFromSystem"}, &comapareTreeCustom, true);
 }
 
 void deploytest::testZIP() {
@@ -713,11 +675,11 @@ void deploytest::testZIP() {
     runTestParams({"-bin", bin, "clear" ,
                    "-qmake", qmake,
                    "-qmlDir", TestBinDir + "/../TestQMLWidgets",
-                   "zip", "verbose"}, &comapareTree, {}, true);
+                   "zip", "verbose"}, &comapareTree, true);
 
 
     // test clear for zip
-    runTestParams({"clear", "verbose"}, {} , {}, true);
+    runTestParams({"clear", "verbose"}, nullptr, true);
 
 }
 
@@ -761,7 +723,7 @@ void deploytest::testZIPMulti() {
                    "-qmlOut", "/q",
                    "-qmlDir", "package2;" + TestBinDir + "/../TestQMLWidgets",
                    "-targetPackage", packageString,
-                   "zip"}, &comapareTreeMulti, {}, true);
+                   "zip"}, &comapareTreeMulti, true);
 }
 
 void deploytest::testDEB() {
@@ -782,10 +744,10 @@ void deploytest::testDEB() {
     runTestParams({"-bin", bin, "clear" ,
                    "-qmake", qmake,
                    "-qmlDir", TestBinDir + "/../TestQMLWidgets",
-                   "deb", "verbose"}, &comapareTree, {}, true);
+                   "deb", "verbose"}, &comapareTree, true);
 
     // test clear for deb
-    runTestParams({"clear", "verbose"}, {} , {}, true);
+    runTestParams({"clear", "verbose"}, nullptr, true);
 
 #endif
 
@@ -823,7 +785,7 @@ void deploytest::testDEBMulti() {
                    "-qmlOut", "/q",
                    "-qmlDir", "package2;" + TestBinDir + "/../TestQMLWidgets",
                    "-targetPackage", packageString,
-                   "deb"}, &comapareTreeMulti, {}, true);
+                   "deb"}, &comapareTreeMulti, true);
 #endif
 }
 
@@ -845,7 +807,7 @@ void deploytest::testDEBCustom() {
                    "-qmlDir", TestBinDir + "/../TestQMLWidgets",
                    "-deb", TestBinDir + "/../../UnitTests/testRes/DEBCustomTemplate",
                    "-name", "chrome"},
-                  &comapareTreeCustom, {}, true);
+                  &comapareTreeCustom, true);
 #endif
 }
 
@@ -871,7 +833,7 @@ void deploytest::testMultiPacking() {
                    "zip",
                    "qif", "qifFromSystem",
                    "deb",
-                   "verbose"}, &comapareTree, {}, true);
+                   "verbose"}, &comapareTree, true);
 
 #else
     auto comapareTree = utils.createTree({
@@ -888,7 +850,7 @@ void deploytest::testMultiPacking() {
                    "-qmlDir", TestBinDir + "/../TestQMLWidgets",
                    "zip",
                    "qif", "qifFromSystem",
-                   "verbose"}, &comapareTree, {}, true);
+                   "verbose"}, &comapareTree, true);
 
 #endif
 }
@@ -1035,7 +997,7 @@ void deploytest::testallowEmptyPackages() {
     QString bin = TestBinDir + "TestOnlyC.exe";
 #endif
     runTestParams({"-bin", bin, "force-clear",
-                   "-prefix", "package;prefix"}, nullptr, {}, false, false,
+                   "-prefix", "package;prefix"}, nullptr, false, false,
                   exitCodes::PrepareError);
 
     runTestParams({"-bin", bin, "force-clear",
@@ -1054,7 +1016,7 @@ void deploytest::testEmptyPackages() {
 #endif
 
     runTestParams({"-bin", bin, "force-clear",
-                   "-prefix", "package;prefix"}, nullptr, {}, false, false,
+                   "-prefix", "package;prefix"}, nullptr, false, false,
                   exitCodes::PrepareError);
 
     runTestParams({"-bin", bin, "force-clear",
@@ -1355,7 +1317,6 @@ void deploytest::testZip() {
 
 void deploytest::runTestParams(QStringList list,
                                QSet<QString>* tree,
-                               const QStringList &checkableKeys,
                                bool noWarnings, bool onlySize,
                                exitCodes exitCode) {
 
@@ -1366,7 +1327,7 @@ void deploytest::runTestParams(QStringList list,
         QVERIFY(false);
 
     if (tree) {
-        checkResults(*tree, checkableKeys, noWarnings, onlySize);
+        checkResults(*tree, noWarnings, onlySize);
     }
 
 #ifdef WITH_SNAP
@@ -1405,7 +1366,6 @@ void deploytest::runTestParams(QStringList list,
 }
 
 void deploytest::checkResults(const QSet<QString> &tree,
-                              const QStringList& checkagbleKeys,
                               bool noWarnings,
                               bool onlySize) {
     TestUtils utils;
@@ -1433,7 +1393,7 @@ void deploytest::checkResults(const QSet<QString> &tree,
                 comapreResult[ i.key()] = "Added unnecessary file";
                 qCritical() << "added unnecessary file : " + i.key();
                 bug = true;
-            } else if (getFilesTree(checkagbleKeys).contains(QFileInfo(i.key()).fileName())) {
+            } else if (filesTree.contains(QFileInfo(i.key()).fileName())) {
                 comapreResult[ i.key()] = "Missing";
                 qCritical() << "Missing file : " + i.key();
                 bug = true;
@@ -1453,7 +1413,7 @@ void deploytest::checkResults(const QSet<QString> &tree,
         }
 
         QJsonObject obj;
-        for (const auto &i : resultTree) {
+        for (const auto &i : qAsConst(resultTree)) {
             obj[i];
         }
 
@@ -2040,12 +2000,12 @@ void deploytest::testIgnore() {
     if (!TestQtDir.contains("Qt5")) {
 
 #ifdef Q_OS_UNIX
-        QString bin = TestBinDir + "QtWidgetsProject";
-        QString qmake = TestQtDir + "bin/qmake";
+        bin = TestBinDir + "QtWidgetsProject";
+        qmake = TestQtDir + "bin/qmake";
 
 #else
-        QString bin = TestBinDir + "QtWidgetsProject.exe";
-        QString qmake = TestQtDir + "bin/qmake.exe";
+        bin = TestBinDir + "QtWidgetsProject.exe";
+        qmake = TestQtDir + "bin/qmake.exe";
 
 #endif
     }
@@ -2194,12 +2154,12 @@ void deploytest::testLibDir() {
     runTestParams({"-bin", bin, "clear" ,
                    "-libDir", extraPath,
                    "-recursiveDepth", "5",
-                   "noCheckRPATH, noCheckPATH", "noQt"}, &comapareTree, {}, true);
+                   "noCheckRPATH, noCheckPATH", "noQt"}, &comapareTree, true);
 
     runTestParams({"-bin", bin, "clear" ,
                    "-targetDir", "./" + DISTRO_DIR + "2",
                    "-extraLibs", "stdc,gcc",
-                   "noCheckRPATH, noCheckPATH", "noQt"}, &comapareTreeExtraLib, {}, true);
+                   "noCheckRPATH, noCheckPATH", "noQt"}, &comapareTreeExtraLib, true);
 
     //task #258
     //https://github.com/QuasarApp/CQtDeployer/issues/258
@@ -2232,7 +2192,7 @@ void deploytest::testLibDir() {
 #endif
     runTestParams({"-bin", bin, "clear" ,
                    "-libDir", extraPath,
-                   "noCheckRPATH, noCheckPATH", "noQt"}, &comapareTreeExtraLib, {}, true);
+                   "noCheckRPATH, noCheckPATH", "noQt"}, &comapareTreeExtraLib, true);
 
     QDir(extraPath).removeRecursively();
 
@@ -2243,7 +2203,6 @@ void deploytest::testExtraPlugins() {
 
 #ifdef Q_OS_UNIX
     QString bin = TestBinDir + "QtWidgetsProject";
-    QString extraPath = "/usr/lib";
     QString qmake = TestQtDir + "bin/qmake";
 
     auto pluginTree = utils.createTree(
@@ -2257,7 +2216,6 @@ void deploytest::testExtraPlugins() {
                 });
 #else
     QString bin = TestBinDir + "QtWidgetsProject.exe";
-    QString extraPath = "/usr/lib";
     QString qmake = TestQtDir + "bin/qmake.exe";
 
     auto pluginTree = utils.createTree(

--- a/UnitTests/tst_deploytest.cpp
+++ b/UnitTests/tst_deploytest.cpp
@@ -20,6 +20,7 @@
 #include <packing.h>
 #include <pluginsparser.h>
 #include <zipcompresser.h>
+#include <QStorageInfo>
 
 #include <QMap>
 #include <QByteArray>
@@ -229,7 +230,7 @@ QStringList deploytest::getFilesFromDir(const QString &path) {
 
     auto list = dir.entryInfoList(QDir::Dirs| QDir::Files| QDir::NoDotAndDotDot);
 
-    for (const auto &subDir: list) {
+    for (const auto &subDir: qAsConst(list)) {
 
         if (subDir.isFile()) {
             res.push_back(subDir.fileName());
@@ -426,7 +427,7 @@ void deploytest::testStrip() {
     };
     QList<qint64> sizeBeforList = {};
 
-    for (auto i: libList) {
+    for (const auto & i: libList) {
         sizeBeforList.push_back(generateLib(i));
     }
 
@@ -435,7 +436,7 @@ void deploytest::testStrip() {
     deploy = new FileManager();
     QVERIFY(deploy->strip("./test/binTargetDir"));
 
-    for(auto i: libList) {
+    for(const auto &i: libList) {
         QFile testLib (i);
         if (testLib.open(QIODevice::ReadOnly)) {
             sizeAfterList.push_back(testLib.size());
@@ -541,7 +542,6 @@ void deploytest::testQIF() {
     TestUtils utils;
 #ifdef Q_OS_UNIX
     QString bin = TestBinDir + "TestQMLWidgets";
-    QString target1 = TestBinDir + "TestOnlyC";
 
     QString qmake = TestQtDir + "bin/qmake";
     auto comapareTree = utils.createTree({
@@ -550,7 +550,6 @@ void deploytest::testQIF() {
 
 #else
     QString bin = TestBinDir + "TestQMLWidgets.exe";
-    QString target1 = TestBinDir + "TestOnlyC.exe";
 
     QString qmake = TestQtDir + "bin/qmake.exe";
     auto comapareTree = utils.createTree({
@@ -578,7 +577,6 @@ void deploytest::testQIFMulti() {
     QString bin = TestBinDir + "TestQMLWidgets";
     QString target1 = TestBinDir + "TestOnlyC";
 
-    QString qmake = TestQtDir + "bin/qmake";
     auto comapareTreeMulti = utils.createTree({
                                                   "./" + DISTRO_DIR + "/InstallerQtWidgetsProject.run",
                                               });
@@ -587,7 +585,6 @@ void deploytest::testQIFMulti() {
     QString bin = TestBinDir + "TestQMLWidgets.exe";
     QString target1 = TestBinDir + "TestOnlyC.exe";
 
-    QString qmake = TestQtDir + "bin/qmake.exe";
     auto comapareTreeMulti = utils.createTree({
                                                   "./" + DISTRO_DIR + "/InstallerQtWidgetsProject.exe",
                                               });
@@ -660,13 +657,11 @@ void deploytest::testZIP() {
 
 #ifdef Q_OS_UNIX
     QString bin = TestBinDir + "TestQMLWidgets";
-    QString target1 = TestBinDir + "TestOnlyC";
 
     QString qmake = TestQtDir + "bin/qmake";
 
 #else
     QString bin = TestBinDir + "TestQMLWidgets.exe";
-    QString target1 = TestBinDir + "TestOnlyC.exe";
 
     QString qmake = TestQtDir + "bin/qmake.exe";
 
@@ -697,7 +692,6 @@ void deploytest::testZIPMulti() {
     QString bin = TestBinDir + "TestQMLWidgets";
     QString target1 = TestBinDir + "TestOnlyC";
 
-    QString qmake = TestQtDir + "bin/qmake";
     QString target2 = TestBinDir + "TestQMLWidgets";
     QString target3 = TestBinDir + "QtWidgetsProject";
 
@@ -707,7 +701,6 @@ void deploytest::testZIPMulti() {
     QString bin = TestBinDir + "TestQMLWidgets.exe";
     QString target1 = TestBinDir + "TestOnlyC.exe";
 
-    QString qmake = TestQtDir + "bin/qmake.exe";
 
 #endif
     bin = target1;
@@ -737,7 +730,6 @@ void deploytest::testDEB() {
 
 
     QString bin = TestBinDir + "TestQMLWidgets";
-    QString target1 = TestBinDir + "TestOnlyC";
 
     QString qmake = TestQtDir + "bin/qmake";
 
@@ -766,8 +758,6 @@ void deploytest::testDEBMulti() {
 
     QString bin = TestBinDir + "TestQMLWidgets";
     QString target1 = TestBinDir + "TestOnlyC";
-
-    QString qmake = TestQtDir + "bin/qmake";
 
     QString target2 = TestBinDir + "TestQMLWidgets";
     QString target3 = TestBinDir + "QtWidgetsProject";
@@ -823,7 +813,6 @@ void deploytest::testMultiPacking() {
                                          });
 
     QString bin = TestBinDir + "TestQMLWidgets";
-    QString target1 = TestBinDir + "TestOnlyC";
 
     QString qmake = TestQtDir + "bin/qmake";
 
@@ -841,7 +830,6 @@ void deploytest::testMultiPacking() {
                                              "./" + DISTRO_DIR + "/InstallerTestQMLWidgets.exe",
                                          });
     QString bin = TestBinDir + "TestQMLWidgets.exe";
-    QString target1 = TestBinDir + "TestOnlyC.exe";
 
     QString qmake = TestQtDir + "bin/qmake.exe";
 
@@ -932,7 +920,7 @@ void deploytest::testQmlScaner() {
 
     QVERIFY(results.size() == imports.size());
 
-    for (auto import: imports) {
+    for (const auto &import: qAsConst(imports)) {
         auto path = scaner->getPathFromImport(import);
         QVERIFY(results.contains(path));
     }
@@ -951,7 +939,7 @@ void deploytest::testQmlScaner() {
 
     QVERIFY(results.size() == imports.size());
 
-    for (auto import: imports) {
+    for (const auto &import: qAsConst(imports)) {
         auto path = scaner->getPathFromImport(import);
         QVERIFY(results.contains(path));
     }
@@ -1076,7 +1064,7 @@ void deploytest::testQmlExtrct() {
     QML scaner("./");
 
 
-    for (const auto &file : qmlFiles) {
+    for (const auto &file : qAsConst(qmlFiles)) {
 
 
         auto fileImports = scaner.extractImportsFromFile(file);
@@ -1115,7 +1103,7 @@ void deploytest::testDistroStruct() {
         {"\\\\res\\\\\\type\\\\\\\\\\","/../../"},
     };
 
-    for (const auto &i: cases) {
+    for (const auto &i: qAsConst(cases)) {
         if (distro.getRelativePath(i.first) != i.second)
             QVERIFY(false);
     }
@@ -1155,7 +1143,7 @@ void deploytest::testRelativeLink() {
 
 };
 
-    for (const auto &i: cases) {
+    for (const auto &i: qAsConst(cases)) {
         if (PathUtils::getRelativeLink(i[0], i[1]) != i[2])
             QVERIFY(false);
     }
@@ -1202,7 +1190,7 @@ void deploytest::testCheckQt() {
 
     };
 
-    for (const auto &i: cases) {
+    for (const auto &i: qAsConst(cases)) {
         QVERIFY(DeployCore::isQtLib(i.first) == i.second);
     }
     delete deployer;
@@ -1264,7 +1252,7 @@ void deploytest::testCheckQt() {
 
     };
 
-    for (const auto &i: cases) {
+    for (const auto &i: qAsConst(cases)) {
         auto dexription = QString("The isQtLib(%0) function should be return %1").arg(
                     i.first).arg(i.second);
         QVERIFY2(DeployCore::isQtLib(i.first) == i.second, dexription.toLatin1().data());
@@ -1374,6 +1362,14 @@ void deploytest::checkResults(const QSet<QString> &tree,
     QVERIFY(!DeployCore::_config->getTargetDir().isEmpty());
 
     auto resultTree = utils.getTree(DeployCore::_config->getTargetDir());
+
+#ifdef Q_OS_WIN
+    // Remove all API-MS-Win libs. becouse each OS Windows have a own bandle of this api libs.
+    // See the https://github.com/QuasarApp/CQtDeployer/issues/481#issuecomment-755156875 post for more information.
+    resultTree = TestModule.ignoreFilter(resultTree, "API-MS-Win");
+
+#endif
+
 
     auto comapre = utils.compareTree(resultTree, tree);
 
@@ -2035,6 +2031,8 @@ void deploytest::testIgnore() {
                                            "./" + DISTRO_DIR + "/lib/libQt5VirtualKeyboard.so",
 
                                        });
+    // ignore all posoble locations with libstdc++-6
+    QString ignreLine = TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools";
 
 #else
     comapareTree = utils.createTree(
@@ -2053,13 +2051,19 @@ void deploytest::testIgnore() {
                                            "./" + DISTRO_DIR + "/plugins/virtualkeyboard/qtvirtualkeyboard_thai.dll"
                                        });
 
+    // Ignore all posoble locations with libstdc++-6 see test log https://github.com/QuasarApp/CQtDeployer/issues/481#issuecomment-755156875
+    // On github actions all gcc libs locate in the program files. So we need to add this path to ignore for testing ignore option.
+    QString winRoot = QStorageInfo::root().rootPath() ;
+    QString programmFiles = winRoot + "/Program Files";
+    QString programmFiles86 = winRoot + "/Program Files (x86)";
 
+    QString ignreLine = TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools" + programmFiles + "," + programmFiles86;
 #endif
 
     runTestParams({"-bin", bin, "clear" ,
                    "-qmake", qmake,
                    "-recursiveDepth", "3",
-                   "-ignoreEnv", TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools" },
+                   "-ignoreEnv",  ignreLine},
                   &comapareTree);
 
 
@@ -2225,6 +2229,7 @@ void deploytest::testExtraPlugins() {
                     "./" + DISTRO_DIR + "/plugins/sqldrivers/qsqlite.dll",
                     "./" + DISTRO_DIR + "/plugins/sqldrivers/qsqlpsql.dll",
                     "./" + DISTRO_DIR + "/Qt5Sql.dll",
+                    "./" + DISTRO_DIR + "/libpq.dll",
 
                 });
 #endif
@@ -2304,80 +2309,6 @@ void deploytest::testSystemLib() {
     auto comapareTree = utils.createTree(
     {
                     "./" + DISTRO_DIR + "/TestOnlyC.exe",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-console-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-datetime-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-debug-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-errorhandling-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-fibers-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-file-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-handle-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-heap-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-localization-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-memory-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-namedpipe-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processenvironment-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processthreads-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-profile-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-string-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-synch-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-synch-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-sysinfo-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-util-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Core-Heap-Obsolete-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Core-Kernel32-Private-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Core-Kernel32-Private-L1-1-1.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-file-l2-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-file-l2-1-1.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-localization-obsolete-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-string-l2-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-string-obsolete-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-xstate-l2-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-com-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-comm-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-datetime-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-debug-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-delayload-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-errorhandling-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-fibers-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-file-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-file-l1-2-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-interlocked-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-io-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-io-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-kernel32-legacy-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-kernel32-legacy-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-libraryloader-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-libraryloader-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-localization-l1-2-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-memory-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-memory-l1-1-2.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-privateprofile-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-privateprofile-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processenvironment-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processthreads-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processthreads-l1-1-2.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processtopology-obsolete-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-realtime-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-registry-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-registry-l2-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-rtlsupport-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-shlwapi-legacy-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-shlwapi-obsolete-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-shutdown-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-stringansi-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-stringloader-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-sysinfo-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-sysinfo-l1-2-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-threadpool-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-threadpool-legacy-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-threadpool-private-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-timezone-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-url-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-version-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-wow64-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-xstate-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-service-core-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-service-core-l1-1-1.dll",
                     "./" + DISTRO_DIR + "/libgcc_s_seh-1.dll",
                     "./" + DISTRO_DIR + "/libstdc++-6.dll",
                     "./" + DISTRO_DIR + "/libwinpthread-1.dll",
@@ -2441,90 +2372,11 @@ void deploytest::testSystemLib() {
 
     comapareTree += utils.createTree(
     {
-                    "./" + DISTRO_DIR + "/api-ms-win-core-console-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-datetime-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-debug-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-errorhandling-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-fibers-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-file-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-handle-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-heap-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-localization-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-memory-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-namedpipe-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processenvironment-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processthreads-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-profile-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-string-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-synch-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-synch-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-sysinfo-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-util-l1-1-0.dll",
                     "./" + DISTRO_DIR + "/libgcc_s_seh-1.dll",
                     "./" + DISTRO_DIR + "/libstdc++-6.dll",
                     "./" + DISTRO_DIR + "/libwinpthread-1.dll",
                     "./" + DISTRO_DIR + "/msvcrt.dll",
                     "./" + DISTRO_DIR + "/qt.conf",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Core-Heap-Obsolete-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Core-Kernel32-Private-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Core-Kernel32-Private-L1-1-1.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Eventing-ClassicProvider-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Eventing-Provider-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-file-l2-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-localization-obsolete-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-string-l2-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-core-string-obsolete-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-devices-config-L1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-delayload-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-file-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-file-l1-2-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-interlocked-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-io-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-io-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-kernel32-legacy-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-kernel32-legacy-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-memory-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-privateprofile-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-processthreads-l1-1-1.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-realtime-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-registry-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-rtlsupport-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-shlwapi-legacy-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-shlwapi-obsolete-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-stringansi-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-sysinfo-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-threadpool-l1-2-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-threadpool-legacy-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-timezone-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-url-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-version-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-core-wow64-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-math-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-private-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-runtime-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-string-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-time-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-security-base-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-security-cryptoapi-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Eventing-Controller-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Eventing-Legacy-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Security-Lsalookup-L2-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-Security-Lsalookup-L2-1-1.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-devices-config-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-security-lsapolicy-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/API-MS-Win-security-provider-L1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-conio-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-convert-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-environment-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-filesystem-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-heap-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-locale-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-multibyte-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-process-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-stdio-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-crt-utility-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-eventing-consumer-l1-1-0.dll",
-                    "./" + DISTRO_DIR + "/api-ms-win-security-sddl-l1-1-0.dll",
                     "./" + DISTRO_DIR + "/mpr.dll",
                     "./" + DISTRO_DIR + "/profapi.dll",
                     "./" + DISTRO_DIR + "/rpcrt4.dll",

--- a/UnitTests/tst_deploytest.cpp
+++ b/UnitTests/tst_deploytest.cpp
@@ -2057,7 +2057,7 @@ void deploytest::testIgnore() {
     QString programmFiles = winRoot + "/Program Files";
     QString programmFiles86 = winRoot + "/Program Files (x86)";
 
-    QString ignreLine = TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools" + programmFiles + "," + programmFiles86;
+    QString ignreLine = TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools," + programmFiles + "," + programmFiles86;
 #endif
 
     runTestParams({"-bin", bin, "clear" ,

--- a/UnitTests/tst_deploytest.cpp
+++ b/UnitTests/tst_deploytest.cpp
@@ -1050,8 +1050,8 @@ void deploytest::testEmptyPackages() {
 }
 
 void deploytest::customTest() {
-    runTestParams({"-confFile", "path",
-                   "qifFromSystem"});
+//    runTestParams({"-confFile", "path",
+//                   "qifFromSystem"});
 }
 
 void deploytest::testQmlExtrct() {
@@ -2031,9 +2031,12 @@ void deploytest::testIgnore() {
                                            "./" + DISTRO_DIR + "/lib/libQt5VirtualKeyboard.so",
 
                                        });
-    // ignore all posoble locations with libstdc++-6
-    QString ignreLine = TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools";
 
+    runTestParams({"-bin", bin, "clear" ,
+                   "-qmake", qmake,
+                   "-recursiveDepth", "3",
+                   "-ignoreEnv",  TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools"},
+                  &comapareTree);
 #else
     comapareTree = utils.createTree(
     {
@@ -2053,19 +2056,18 @@ void deploytest::testIgnore() {
 
     // Ignore all posoble locations with libstdc++-6. See tests log https://github.com/QuasarApp/CQtDeployer/issues/481#issuecomment-755156875 for more information.
     // On github actions all gcc libs locate in the program files. So we need to add this path to ignore for testing ignore option.
-    QString winRoot = QStorageInfo::root().rootPath() ;
+    QString winRoot = QStorageInfo::root().rootPath();
     QString programmFiles = winRoot + "/Program Files";
     QString programmFiles86 = winRoot + "/Program Files (x86)";
-
-    QString ignreLine = TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools," + programmFiles + "," + programmFiles86;
-#endif
 
     runTestParams({"-bin", bin, "clear" ,
                    "-qmake", qmake,
                    "-recursiveDepth", "3",
-                   "-ignoreEnv",  ignreLine},
+                   "-ignoreEnv",  TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools",
+                   "-ignore", programmFiles + "," + programmFiles86},
                   &comapareTree);
 
+#endif
 
     comapareTree = TestModule.qtLibs() - removeTree;
 

--- a/UnitTests/tst_deploytest.cpp
+++ b/UnitTests/tst_deploytest.cpp
@@ -1364,7 +1364,7 @@ void deploytest::checkResults(const QSet<QString> &tree,
     auto resultTree = utils.getTree(DeployCore::_config->getTargetDir());
 
 #ifdef Q_OS_WIN
-    // Remove all API-MS-Win libs. becouse each OS Windows have a own bandle of this api libs.
+    // Remove all API-MS-Win libs, because each OS Windows have a own bundle of this api libs.
     // See the https://github.com/QuasarApp/CQtDeployer/issues/481#issuecomment-755156875 post for more information.
     resultTree = TestModule.ignoreFilter(resultTree, "API-MS-Win");
 
@@ -2051,7 +2051,7 @@ void deploytest::testIgnore() {
                                            "./" + DISTRO_DIR + "/plugins/virtualkeyboard/qtvirtualkeyboard_thai.dll"
                                        });
 
-    // Ignore all posoble locations with libstdc++-6 see test log https://github.com/QuasarApp/CQtDeployer/issues/481#issuecomment-755156875
+    // Ignore all posoble locations with libstdc++-6. See tests log https://github.com/QuasarApp/CQtDeployer/issues/481#issuecomment-755156875 for more information.
     // On github actions all gcc libs locate in the program files. So we need to add this path to ignore for testing ignore option.
     QString winRoot = QStorageInfo::root().rootPath() ;
     QString programmFiles = winRoot + "/Program Files";

--- a/UnitTests/tst_deploytest.cpp
+++ b/UnitTests/tst_deploytest.cpp
@@ -58,6 +58,9 @@ private:
     void checkResults(const QSet<QString> &tree,
                       bool noWarnings,
                       bool onlySize = false);
+
+    void createTree(const QStringList& tree);
+
 public:
     deploytest();
     /**
@@ -115,8 +118,11 @@ private slots:
     // tested clear force clear in clear mode
     void testClear();
 
-    // tested flags ignore ignoreEnv
+    // tested flags ignore
     void testIgnore();
+
+    // tested flags ignore ignoreEnv
+    void testIgnoreEnv();
 
     // tested flags libDir recursiveDepth
     void testLibDir();
@@ -1434,6 +1440,12 @@ void deploytest::checkResults(const QSet<QString> &tree,
 
 }
 
+void deploytest::createTree(const QStringList &tree) {
+    for (const auto& dir : tree) {
+        QDir().mkpath(dir);
+    }
+}
+
 void deploytest::costomScript() {
     TestUtils utils;
 
@@ -2014,13 +2026,6 @@ void deploytest::testIgnore() {
 
 
 #ifdef Q_OS_UNIX
-    comapareTree = utils.createTree(
-    {
-                    "./" + DISTRO_DIR + "/QtWidgetsProject.sh",
-                    "./" + DISTRO_DIR + "/bin/qt.conf",
-                    "./" + DISTRO_DIR + "/bin/QtWidgetsProject",
-                });
-
     auto removeTree = utils.createTree({
                                            "./" + DISTRO_DIR + "/plugins/virtualkeyboard/libqtvirtualkeyboard_hangul.so",
                                            "./" + DISTRO_DIR + "/plugins/virtualkeyboard/libqtvirtualkeyboard_openwnn.so",
@@ -2031,19 +2036,7 @@ void deploytest::testIgnore() {
                                            "./" + DISTRO_DIR + "/lib/libQt5VirtualKeyboard.so",
 
                                        });
-
-    runTestParams({"-bin", bin, "clear" ,
-                   "-qmake", qmake,
-                   "-recursiveDepth", "3",
-                   "-ignoreEnv",  TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools"},
-                  &comapareTree);
 #else
-    comapareTree = utils.createTree(
-    {
-                    "./" + DISTRO_DIR + "/qt.conf",
-                    "./" + DISTRO_DIR + "/QtWidgetsProject.exe",
-                });
-
     auto removeTree = utils.createTree({
                                            "./" + DISTRO_DIR + "/Qt5VirtualKeyboard.dll",
                                            "./" + DISTRO_DIR + "/plugins/platforminputcontexts/qtvirtualkeyboardplugin.dll",
@@ -2054,19 +2047,6 @@ void deploytest::testIgnore() {
                                            "./" + DISTRO_DIR + "/plugins/virtualkeyboard/qtvirtualkeyboard_thai.dll"
                                        });
 
-    // Ignore all posoble locations with libstdc++-6. See tests log https://github.com/QuasarApp/CQtDeployer/issues/481#issuecomment-755156875 for more information.
-    // On github actions all gcc libs locate in the program files. So we need to add this path to ignore for testing ignore option.
-    QString winRoot = QStorageInfo::root().rootPath();
-    QString programmFiles = winRoot + "/Program Files";
-    QString programmFiles86 = winRoot + "/Program Files (x86)";
-
-    runTestParams({"-bin", bin, "clear" ,
-                   "-qmake", qmake,
-                   "-recursiveDepth", "3",
-                   "-ignoreEnv",  TestQtDir + "/lib," + TestQtDir + "/bin," + TestQtDir + "/../../Tools",
-                   "-ignore", programmFiles + "," + programmFiles86},
-                  &comapareTree);
-
 #endif
 
     comapareTree = TestModule.qtLibs() - removeTree;
@@ -2075,6 +2055,50 @@ void deploytest::testIgnore() {
                    "-qmake", qmake,
                    "-ignore", "VirtualKeyboard"}, &comapareTree);
 
+}
+
+void deploytest::testIgnoreEnv() {
+
+
+
+    Envirement env;
+    QDir("./testTree").removeRecursively();
+
+    QStringList ignoreTree = {
+        "./testTree/test",
+        "./testTree/",
+        "./testTree/test1/1",
+        "./testTree/test2/1/",
+    };
+
+    QStringList testTree = {
+        "./testTree/test/z",
+        "./testTree/z",
+        "./testTree/test1/1z",
+        "./testTree/test2/1/z",
+    };
+
+    createTree(ignoreTree);
+    createTree(testTree);
+
+    env.setIgnoreEnvList(ignoreTree);
+    env.addEnv(ignoreTree);
+
+    // must be empty becouse all pathes is ignored
+    QVERIFY(env.size() == 0);
+
+    env.addEnv(testTree);
+
+    // must be equals 4 becouse all pathes is not ignored
+    QVERIFY(env.size() == 4);
+
+    // try add dublicate
+    env.addEnv(testTree);
+
+    // must be equals 4 becouse all dublicates must be ignored
+    QVERIFY(env.size() == 4);
+
+    QVERIFY(QDir("./testTree").removeRecursively());
 }
 
 void deploytest::testLibDir() {
@@ -2395,6 +2419,7 @@ void deploytest::testSystemLib() {
 
                 });
 
+    "./" + DISTRO_DIR + "/libwinpthread-1.dll";
 
 #if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
     comapareTree += utils.createTree(

